#!/bin/sh -e
## by Emily Dai <emily.dai@linspireinc.com>
##
## DP: Fixed file/image saving problem

if [ $# -ne 1 ]; then
echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
	-patch) patch $patch_opts -p1 < $0;;
	-unpatch) patch $patch_opts -p1 -R < $0;;
	*)
		echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
		exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /var/tmp/gpaint-0.2.3/src/fileio.c gpaint-0.2.3/src/fileio.c
--- /var/tmp/gpaint-0.2.3/src/fileio.c	2002-11-14 06:50:57.000000000 -0200
+++ gpaint-0.2.3/src/fileio.c	2004-09-17 08:17:18.000000000 -0300
@@ -44,9 +44,16 @@
     for (i = 0; i <strlen(type); i++)
         type[i] = tolower(type[i]);
         
-    if (!strcmp(ctype, "jpg"))
+    if (!strcmp(type, "jpg"))
         strncpy(type, "jpeg", sizeof(type));
-    
+
+    ibuf->rgbbuf = gdk_pixbuf_get_from_drawable(NULL,
+						GDK_DRAWABLE(ibuf->pixmap),
+						NULL,
+						0, 0,
+						0, 0,
+						image_buf_width(ibuf),
+						image_buf_height(ibuf));
     if (!strcmp(type, "jpeg"))    
         i = gdk_pixbuf_save(ibuf->rgbbuf, name, type, &gerr, "quality", "100", 0);
     else
